ledger/src/shred.rs

```diff
@@ -386,24 +342,14 @@ impl Shred {
             position,
         };
         let mut payload = vec![0; SHRED_PAYLOAD_SIZE];
-        let mut start = 0;
-        Self::serialize_obj_into(
-            &mut start,
-            SIZE_OF_COMMON_SHRED_HEADER,
-            &mut payload,
-            &common_header,
-        )
-        .expect("Failed to write header into shred buffer");
-        Self::serialize_obj_into(
-            &mut start,
-            SIZE_OF_CODING_SHRED_HEADER,
-            &mut payload,
-            &coding_header,
-        )
-        .expect("Failed to write coding header into shred buffer");
+        let mut cursor = Cursor::new(&mut payload[..]);
+        bincode::serialize_into(&mut cursor, &common_header).unwrap();
+        bincode::serialize_into(&mut cursor, &coding_header).unwrap();
         // Tests may have an empty parity_shard.
         if !parity_shard.is_empty() {
-            payload[SIZE_OF_CODING_SHRED_HEADERS..].copy_from_slice(parity_shard);
+            let offset = cursor.position() as usize;
+            debug_assert_eq!(offset, SIZE_OF_CODING_SHRED_HEADERS);
```

Comment:
Same as above - think this could be in a unit test

Reply:
similarly here https://github.com/solana-labs/solana/pull/24876/files#r862835368